src/poll.rs

```diff
@@ -68,95 +113,155 @@ pub struct Registration {
     inner: RegistrationInner,
 }
 
+unsafe impl Send for Registration {}
+unsafe impl Sync for Registration {}
+
 /// Used to update readiness for an associated `Registration`. `SetReadiness`
 /// is `Sync` which allows it to be updated across threads.
 #[derive(Clone)]
 pub struct SetReadiness {
     inner: RegistrationInner,
 }
 
+unsafe impl Send for SetReadiness {}
+unsafe impl Sync for SetReadiness {}
+
 struct RegistrationInner {
     // ARC pointer to the Poll's readiness queue
     queue: ReadinessQueue,
 
-    // Unsafe pointer to the registration's node. The node is owned by the
-    // registration queue.
-    node: ReadyRef,
+    // Unsafe pointer to the registration's node. The node is ref counted. This
+    // cannot "simply" be tracked by an Arc because `Poll::poll` has an implicit
+    // handle though it isn't stored anywhere. In other words, `Poll::poll`
+    // needs to decrement the ref count before the node is freed.
+    node: *mut ReadinessNode,
 }
 
 #[derive(Clone)]
 struct ReadinessQueue {
     inner: Arc<UnsafeCell<ReadinessQueueInner>>,
```

Comment:
Is this `UnsafeCell` required? It looks like `ReadinessQueueInner` already has all the proper mutability related to `UnsafeCell` and such, so this seems like it should be able to be `Arc<ReadinessQueueInner>`